### PR TITLE
bump 02-client-refactor branch to tip of main

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by colin-axner** [comment]
still need to fix broken tests

File: docs/migrations/v2-to-v3.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ There are four sections based on the four potential user groups of this document
 
 **Note:** ibc-go supports golang semantic versioning and therefore all imports must be updated to bump the version number on major releases.
 ```go
-github.com/cosmos/ibc-go/v2 -> github.com/cosmos/ibc-go/v3
+github.com/cosmos/ibc-go/v2 -> github.com/cosmos/ibc-go/v5
```
Comment (damiannolan) [inline docs/migrations/v2-to-v3.md:14]:
```suggestion
github.com/cosmos/ibc-go/v2 -> github.com/cosmos/ibc-go/v3
```

File: docs/migrations/v3-to-v4.md [hunk 2]
```diff
@@ -11,16 +11,122 @@ There are four sections based on the four potential user groups of this document
 
 **Note:** ibc-go supports golang semantic versioning and therefore all imports must be updated to bump the version number on major releases.
 ```go
-github.com/cosmos/ibc-go/v3 -> github.com/cosmos/ibc-go/v4
+github.com/cosmos/ibc-go/v5 -> github.com/cosmos/ibc-go/v4
```
Comment (damiannolan) [inline docs/migrations/v3-to-v4.md:14]:
```suggestion
github.com/cosmos/ibc-go/v3 -> github.com/cosmos/ibc-go/v4
```

**Review by damiannolan — COMMENTED** [review]
Github broken after first 78 files for me. Just left some comments on import paths and changelog so far 👍

File: CHANGELOG.md [hunk 3]
```diff
@@ -38,25 +38,40 @@ Ref: https://keepachangelog.com/en/1.0.0/
...
 
 ### API Breaking
 
-* (transfer) [\#1250](https://github.com/cosmos/ibc-go/pull/1250) Deprecate `GetTransferAccount` since the `transfer` module account is never used.
 * (06-solomachine) [\#1679](https://github.com/cosmos/ibc-go/pull/1679) Remove `types` sub-package from `06-solomachine` lightclient directory. 
 * (07-tendermint) [\#1677](https://github.com/cosmos/ibc-go/pull/1677) Remove `types` sub-package from `07-tendermint` lightclient directory. 
 * (06-solomachine) [\#1687](https://github.com/cosmos/ibc-go/pull/1687) Bump `06-solomachine` protobuf version from `v2` to `v3`.
 * (06-solomachine) [\#1687](https://github.com/cosmos/ibc-go/pull/1687) Removed `DataType` enum and associated message types from `06-solomachine`. `DataType` has been removed from `SignBytes` and `SignatureAndData` in favour of `path`.
+* [\#1615](https://github.com/cosmos/ibc-go/pull/1615) Bump SDK version to v0.45.6
+* [\#1627](https://github.com/cosmos/ibc-go/pull/1627) Bump Go version to 1.18
```
[the remark this answers is not in the document]
Comment (seantking) [reply CHANGELOG.md:49]: The 0.45.6 bump is in the current dependencies for the unreleased section (changelog on main). 

I think we didn't add a changelog entry for the bump to SDK 46 yet.

[the remark this answers is not in the document]
Comment (colin-axner) [reply CHANGELOG.md]: thanks for catching all these issues! Applied fixes

**Comment by colin-axner** [comment]
merging into 02-client-refactor as that should be easier to review against main

**Pull request merged**